.github/workflows/validate-ena-flags.yaml

```
@@ -0,0 +1,31 @@
+name: Validate submitToEnaProduction is not true in values.yaml
+on:
+    pull_request:
+    push:
+        branches:
+        - main
+concurrency:
+  group: ci-${{ github.ref == 'refs/heads/main' && github.run_id || github.ref }}-validate-ena-flags
+  cancel-in-progress: true
+jobs:
+  validate-ena-flags:
+    name: Validate submitToEnaProduction is not true in values.yaml # Don't change: Referenced by .github/workflows/update-argocd-metadata.yml
+    permissions:
+      packages: read
+      contents: read
+      checks: read
+    runs-on: ubuntu-latest
+    timeout-minutes: 45
+    env:
+      sha: ${{ github.event.pull_request.head.sha || github.sha }}
+      wait_timeout: ${{ github.ref == 'refs/heads/main' && 900 || 240 }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Validate submitToEnaProduction is not true in values.yaml
+        run: |
+          SUBMIT_TO_ENA_PROD=$(grep 'submitToEnaProduction:' kubernetes/loculus/values.yaml | awk '{print $2}')
+          if [[ "$SUBMIT_TO_ENA_PROD" == "true" ]]; then
```

```suggestion
          if [[ "$SUBMIT_TO_ENA_PROD" == "true" ]]; then
```

I think this isn't great, you should use `yq` to check the truthiness here rather than look for an exact string match as many values can evaluate to true in yaml, more things than `true` for sure - e.g. `True` is likely also true

---

ChatGPT suggests the following, which is better as it uses the same library to validate the truthyness as the eventual program that reads that config value:

```
      - name: Validate submitToEnaProduction is not true in values.yaml
        run: |
          python -c "
import yaml
with open('kubernetes/loculus/values.yaml', 'r') as file:
    values = yaml.safe_load(file)
    submit_to_ena_prod = values.get('submitToEnaProduction', False)
    if submit_to_ena_prod:
        print('Error: The flag \'submitToEnaProduction\' is set to true - this will submit data to ENA production. Please set it to false in values.yaml')
        exit(1)
"

---

cool - I couldn't install `yq` for the check but maybe this works -> lets try!